### DISCUSSION TITLE
Cannot POST 1 to /listings (V3 endpt) with SKU

### DISCUSSION THREAD

**Discussion by nathanp-rs** [etsy/open-api#1130] [Q&A]
Hello, 

I'm attempting to post a new product with the following payload:
```
        let payload = {
            "sku": "test-sku",
            "quantity": qty,
            "title": title,
            "description": description,
            "price": price,
            "who_made": "someone_else",
            "when_made": when_made,
            "taxonomy_id": taxonomyId,
            "section_id": sectionId,
            "shipping_profile_id": "111230952426"
          };
```
The response comes back with all data correct except a SKU added with no error messages or warning, just a different key-value called `skus: []`.

I can also confirm that the v3 `PATCH` `/listings/:listingId` passing a SKU into the request body also does not work as expected. 

In this case with endpt: `https://api.etsy.com/v3/application/shops/25519898/listings/1571883352`

Pertaining to the V3 endpoints, can anyone confirm SKU is a parameter that can be passed thru at all?

[createDraftListing](https://developer.etsy.com/documentation/reference#operation/createDraftListing)

[updateListing](https://developer.etsy.com/documentation/reference/#operation/updateListing)

SKU is not mentioned in the Body Schema for either of these endpoints, is it a field that can be programmatically altered at all or not? And if not why not and is this supported some other way?

Creating listings with SKUs is a business critical requirement for us. It would be very surprising if this field is not supported, Etsy Dev support has also been unresponsive. 

Has anyone else encountered this?

Thanks for your input!

**Comment by leedyche** [comment]
Ive come against this same issue, i am unable to find a way to set the sku field when creating listings or updating them. Did you get to the bottom of this?

**Reply by valigara** [reply]
on listing level you can't set SKU, to do this use [updateListingInventory](https://developers.etsy.com/documentation/reference#operation/updateListingInventory)

flow
- [createDraftListing](https://developer.etsy.com/documentation/reference#operation/createDraftListing) - base information
- [updateListingInventory](https://developers.etsy.com/documentation/reference#operation/updateListingInventory) - skus & variations
- add at least one image, because etsy requires at least one image on listing to activate it
- [updateListing](https://developer.etsy.com/documentation/reference/#operation/updateListing) - to activate listing

**Comment by showtroylove** [comment]
See my previous response to this issue.  You must perform a full **POST** as one entire (giant record).  You can't target (**PATCH**) an individual SKU, Price, Offering, etc.  

**Be sure to mark this comment as an answer if it helps you resolve your question.**
